sklearn/model_selection/_search.py

```diff
@@ -727,10 +728,17 @@ def evaluate_candidates(candidate_params):
         # best_score_ iff refit is one of the scorer names
         # In single metric evaluation, refit_metric is "score"
         if self.refit or not self.multimetric_:
-            self.best_index_ = results["rank_test_%s" % refit_metric].argmin()
+            # If self.refit is a callable, it will be passed results and would
```

Comment:
Or maybe "If callable, `refit` is expected to return the index of the best parameter set."